env.sh

```diff
@@ -1,4 +1,21 @@
-CURRENT=$(dirname $0)
+set -e
+
+# https://stackoverflow.com/a/28776166
+is_sourced() {
+    if [ -n "$ZSH_VERSION" ]; then
+        case $ZSH_EVAL_CONTEXT in *:file:*) return 0;; esac
+    else  # Add additional POSIX-compatible shell names here, if needed.
+        case ${0##*/} in dash|-dash|bash|-bash|ksh|-ksh|sh|-sh) return 0;; esac
+    fi
+    return 1  # NOT sourced.
+}
+
+if ! is_sourced || ! [ -f $PWD/env.sh ] ; then
```

Comment:
This looks unnecessary to me given the complexity it brings. It should be okay to run the script without sourcing it, right? It'll just print variables and the user will be able to decide what they do with that.

Reply:
We could change the script to do the printenv directly, but it's not what it's doing right now, so it's a bit unrelated.

Reply:
Oh, right, this is unrelated :facepalm: